openproblems/tasks/spatial_decomposition/datasets/_utils.py

```diff
@@ -0,0 +1,88 @@
+from .._utils import obs_means
+from anndata import AnnData
+
+import numpy as np
+import random
+
+
+def generate_seq(length=14):
+    return "".join(random.choice("CGTA") for _ in range(length))
+
+
+# pass the reference data
+def generate_synthatic_data(adata, sim_type="avg"):
+    n_genes = adata.n_var
+    n_cells = adata.n_obs
+    n_types = len(set(adata.obs["label"].values))
+
+    # TODO(make these arguments)
+    bead_depth = 1000
+    num_of_beads = n_cells * 2
+    # generate proportion values
+    props = np.random.dirichlet(np.ones(n_types), num_of_beads)
```

Comment:
would be good to set all random seeds at the beginning and also use the new `random_generator` from numpy: https://numpy.org/doc/stable/reference/random/generator.html

Reply:
basically idea is that all random generators have same seed and is set only once at init

Reply:
I think this would be great to have, but haven't worked with it too much myself; maybe something we could implement later on?